internal/route/api/v1/repo/contents.go

```diff
@@ -0,0 +1,177 @@
+package repo
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+
+	"github.com/gogs/git-module"
+
+	"gogs.io/gogs/internal/context"
+)
+
+type repoContents struct {
```

Comment:
Sorry you're right, I didn't notice when requesting a directory, we should return `[]*repoContent`.
```suggestion
type repoContent struct {
```

Reply:
Fixed, thanks!